src/palette.ml

```
@@ -43,6 +43,27 @@ let load_tic80_palette (raw : string) : t =
   else
     raise (Invalid_argument "Palette size must not be zero or negative")
 
+(* New function for Task #27 *)
+let load_lospec_palette (s : string) : t =
+  let lines = String.split_on_char '\n' s in
+  let parse_hex line =
+    let line = String.trim line in
+    let hex =
+      if String.length line = 6 then line
+      else if String.length line = 7 && line.[0] = '#' then String.sub line 1 6
+      else ""
+    in
+    if hex <> "" then
+      match int_of_string_opt ("0x" ^ hex) with
+      | Some n -> Some n
+      | None -> None
```

I think if there's any errors parsing the file it's best to fail. If I have a large palette, it may not be obvious we missed a few items, and it'd be a pain to debug. We should fail in a way that makes the error obvious.

---

I have added explicit error handling for malformed lines and empty palettes and also removed reliance on filter_map to avoid silent errors.